login/src/commonMain/kotlin/org/dhis2/mobile/login/authentication/ui/screen/TwoFAToEnableScreen.kt

```diff
@@ -0,0 +1,292 @@
+import org.dhis2.mobile.login.resources.two_fa_to_enable_google_authenticator
+import org.dhis2.mobile.login.resources.two_fa_to_enable_step_one_description
+import org.dhis2.mobile.login.resources.two_fa_to_enable_step_one_number
+import org.dhis2.mobile.login.resources.two_fa_to_enable_step_three_description
+import org.dhis2.mobile.login.resources.two_fa_to_enable_step_three_number
+import org.dhis2.mobile.login.resources.two_fa_to_enable_step_two_description
+import org.dhis2.mobile.login.resources.two_fa_to_enable_step_two_number
+import org.dhis2.mobile.login.resources.two_fa_turn_on_button
+import org.dhis2.mobile.login.resources.two_fa_turn_on_title
+import org.dhis2.mobile.login.resources.two_fa_turning_on_button
+import org.hisp.dhis.mobile.ui.designsystem.component.Button
+import org.hisp.dhis.mobile.ui.designsystem.component.ButtonStyle
+import org.hisp.dhis.mobile.ui.designsystem.component.IconButton
+import org.hisp.dhis.mobile.ui.designsystem.component.IconButtonStyle
+import org.hisp.dhis.mobile.ui.designsystem.component.InfoBar
+import org.hisp.dhis.mobile.ui.designsystem.component.InputShellState
+import org.hisp.dhis.mobile.ui.designsystem.component.InputText
+import org.hisp.dhis.mobile.ui.designsystem.component.SupportingTextData
+import org.hisp.dhis.mobile.ui.designsystem.component.SupportingTextState
+import org.hisp.dhis.mobile.ui.designsystem.theme.Radius
+import org.hisp.dhis.mobile.ui.designsystem.theme.Spacing
+import org.hisp.dhis.mobile.ui.designsystem.theme.SurfaceColor
+import org.hisp.dhis.mobile.ui.designsystem.theme.TextColor
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+
+@Composable
+fun TwoFAToEnableScreen(
+    enableUiState: TwoFaEnableUiState,
+    secretCode: String,
+    onAuthenticatorButtonClicked: () -> Unit,
+    onCopyCodeButtonClicked: (String) -> Unit,
+    onEnableButtonClicked: (String) -> Unit,
+) {
+    Column(
+        verticalArrangement = spacedBy(Spacing.Spacing12),
+    ) {
+        InfoBar(
+            text = stringResource(Res.string.two_fa_status_off),
+            textColor = TextColor.OnWarningContainer,
+            backgroundColor = SurfaceColor.WarningContainer,
+            icon = {
+                Icon(
+                    imageVector = Icons.Outlined.KeyOff,
+                    contentDescription = stringResource(Res.string.two_fa_no_connection_title),
+                    tint = TextColor.OnWarningContainer,
+                )
+            },
+        )
+
+        Text(
+            text = stringResource(Res.string.two_fa_turn_on_title),
+            style = MaterialTheme.typography.titleMedium,
+        )
+
+        TwoFAAuthStepOne(onAuthenticatorButtonClicked)
+
+        TwoFAAuthStepTwo(secretCode, onCopyCodeButtonClicked)
+
+        TwoFAAuthStepThree(enableUiState, onEnableButtonClicked)
+    }
+}
+
+@Composable
+fun TwoFAAuthStepOne(onAuthenticatorButtonClicked: () -> Unit) {
```

Comment:
Do you think that we could find more self explained naming for this screen states than `TwoFAAuthStepOne` or `TwoFAAuthStepTwo`?